tests/trade_executor_tests.cc

```diff
@@ -168,22 +162,124 @@ TEST_F(TradeExecutorTests, NoThrowWriting) {
   EXPECT_NO_THROW(exec.RecordTrades(tc.get()));
 }
 
-// This test was a part of a previous iteration of Trade testing, but its not
-// clear if this throwing behavior is what we want. I'm leaving it here for now
-// in case it needs to be picked up again. MJG - 11/26/13
-// // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-// TEST(TradeTests, OfferThrow) {
-//   TestContext tc;
-
-//   Material::Ptr mat = get_mat();
-//   Receiver* r = new Receiver(tc.get(), mat);
-//   Request<Material>* req = Request<Material>::Create(mat, r);
-
-//   Sender* s = new Sender(tc.get(), true);
-//   Bid<Material>* bid = Bid<Material>::Create(req, mat, s);
-
-//   Trade<Material> trade(req, bid, mat->quantity());
-//   EXPECT_THROW(cyclus::ExecuteTrade(trade), cyclus::ValueError);
-//   delete s;
-//   delete r;
-// }
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+class SelfTradingWarningTest : public ::testing::Test {
+ public:
+  virtual void SetUp() {
+
+    tc_ = std::make_unique<TestContext>();
+    
+    facility_ = new SelfTradingTestFacility(tc_->get());
+    
+    facility_->Build(nullptr);
+    facility_->EnterNotify();
+    
+    CompMap v;
+    v[id("u235")] = 1;
+    double trade_amt = 100;
+
+    test_comp_ = Composition::CreateFromAtom(v);
+    test_mat_ = Material::CreateUntracked(trade_amt, test_comp_);
+  }
+
+  virtual void TearDown() {
+    delete facility_;
+  }
+
+ protected:
+  std::unique_ptr<TestContext> tc_;
+  SelfTradingTestFacility* facility_;
+  Composition::Ptr test_comp_;
+  Material::Ptr test_mat_;
+  
+  // Test constants
+  static constexpr double kTestTradeAmount = 50.0;
+};
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+TEST_F(SelfTradingWarningTest, SelfTradingWarningIssued) {
+  warn_as_error = true;
```

Comment:
Maybe do this as late as possible, I.e. immediately before the call that throws the warning